packages/next/src/middleware/request.ts

```diff
@@ -5,16 +5,9 @@ import type { NextRequest as InternalNextRequest } from 'next/server'
 import { MiddlewareResponse } from './response'
 
 export type NextRequest = InternalNextRequest & {
-  get geo():
-    | {
-        city?: string
-        country?: string
-        region?: string
-        latitude?: string
-        longitude?: string
-        timezone?: string
-      }
-    | undefined
+  get geo(): {
+    timezone?: string
```

Comment:
Oh, I didn't realise it would inherit by default!

Reply:
Yeah that what I thought you meant initially and not an intersection, but it looks like it works. 😎